src/chains.jl

```diff
@@ -783,10 +796,15 @@ function _cat(::Val{2}, c1::Chains, args::Chains...)
     data = mapreduce(c -> c.value.data, hcat, args; init = c1.value.data)
     value = AxisArray(data; iter = rng, var = nms, chain = chns)
 
-    return Chains(value, missing, name_map, c1.info)
+    return Chains(value, missing, name_map, c1.info, c1.weights)
 end
 
 function _cat(::Val{3}, c1::Chains, args::Chains...)
+    # TODO: Deal with the case that some chains have non-unit weights.
+    # Requires implementing array-like weights for StatsBase (#748)
+    if !all(c -> typeof(c.weights)<:UnitWeights, args)
+        throw(ArgumentError("all chains must have unit weights to concatenate"))
+    end
```

Comment:
This seems too restrictive? I guess it is sufficient if they are equal.
```suggestion
    # check weights
    w = c1.weights
    all(c -> c.weights == w, args) || throw(ArgumentError("all chains must have unit weights to concatenate"))
```

Reply:
Is there any situation where we want to concatenate chains with non-unit weights?

Reply:
I don't know - but not even unit weights have to be represented as `UnitWeights`, e.g., it could also be of type `Weights` with an array of equal weights. Not checking the types but the values (which uses shortcuts for special types such as `UnitWeights`: https://github.com/JuliaStats/StatsBase.jl/blob/071d10a110d756fa15c6a6878936eee27e4c2cb6/src/weights.jl#L369) works much more generally and also with weight types not defined in StatsBase.